--- a/configure
+++ b/configure
@@ -906,6 +906,7 @@ datarootdir
 libexecdir
 sbindir
 bindir
+stagingdir
 program_transform_name
 prefix
 exec_prefix
@@ -1040,6 +1041,7 @@ with_suexec_gidmin
 with_suexec_logfile
 with_suexec_safepath
 with_suexec_umask
+with_staging_dir
 '
       ac_precious_vars='build_alias
 host_alias
@@ -21610,6 +21612,14 @@ _ACEOF
 
 fi
 
+# Check whether --with-staging-dir was given.
+if test "${with_staging_dir+set}" = set; then
+  withval=$with_staging_dir;
+  stagingdir="$(echo $withval | sed 's/\/*$//')"
+else
+  stagingdir=""
+fi
+
 
 apulinklibs="`$apu_config --avoid-ldap --link-libtool --libs`" \
  || apulinklibs="`$apu_config --link-libtool --libs`"
